### PR TITLE
Enhance Widget Layout

### PR DESCRIPTION
**Pull request by Spraynard**
This update contains two changes to two different files.

I had an issue similar to Issue #3185 and was able to implement a solution that works along with the current voyager configuration but allow for more than three widgets to be on the dashboard window.

This is what the dashboard window looks like with 1 - 3 Widgets

voyager_widgets_1

This is what the dashboard window looks like with 5 widgets

voyager_widgets_2

Please let me know if there is any more information needed

### CODE REVIEW

File: tests/Unit/VoyagerTest.php [hunk 1]
```diff
@@ -42,6 +42,27 @@ public function testDimmersReturnsCollectionOfConfiguredWidgetsWhichShouldBeDisp
...
+        $this->assertEquals(1, $dimmers[0]->count());
+    }
+
+    /**
+     * Dimmers returns an array filled with widget collections.
+     *
+     * This test ensures that each widget collection has a maximum of three widgets/dimmers
+     * creating as many groups as are needed in order to encompass the full range of dimmers.
+     */
+    public function testEachDimmerGroupHasAMaxAmountOfThreeDimmers()
```
Comment (MrCrayon) [inline tests/Unit/VoyagerTest.php]: If you want to test that Group has max amount of three Dimmers you should set 3 dimmers in config and assert equals 1, then set 4 dimmers and assert equals 2.

Right now you could have 4 dimmers in the first group without error.

**Review by MrCrayon — CHANGES_REQUESTED** [review]
Except Test everything else looks good. :+1:

File: tests/Unit/VoyagerTest.php (hunk 1 above)
Comment (Spraynard) [reply tests/Unit/VoyagerTest.php]: Thank you for the guidance. I'll commit an updated VoyagerTest.php

File: tests/Unit/VoyagerTest.php [hunk 2]
```diff
@@ -64,5 +83,7 @@ public function testEachDimmerGroupHasAMaxAmountOfThreeDimmers()
         $dimmers = Voyager::dimmers();
 
         $this->assertEquals(2, count($dimmers));
+        $this->assertEquals(3, $dimmers[0]->count());
```
Comment (MrCrayon) [inline tests/Unit/VoyagerTest.php:67]: This assertion proves what you were trying to prove in the previous test, so I think the other one can be removed.

**Review by MrCrayon — CHANGES_REQUESTED** [review]
I think you can remove the first test.

File: tests/Unit/VoyagerTest.php [hunk 3]
```diff
@@ -42,6 +42,48 @@ public function testDimmersReturnsCollectionOfConfiguredWidgetsWhichShouldBeDisp
...
-        $this->assertEquals(1, $dimmers->count());
+        $this->assertEquals(1, $dimmers[0]->count());
+    }
+
+    /**
+     * Dimmers returns an array filled with widget collections.
+     *
+     * This test ensures that each widget collection can hold up to three widgets
+     */
+    public function testDimmerCollectionContainsThreeDimmersMax()
```
Comment (MrCrayon) [inline tests/Unit/VoyagerTest.php]: How can you test that collection contains three dimmers max if you are only passing three? :thinking:

**Review by MrCrayon — CHANGES_REQUESTED** [review]
This test can be removed.

**Review by MrCrayon — APPROVED** [review]
Looks good, thanks for this PR

**Comment by Spraynard** [comment]
Thank you very much for the help as well. It's much appreciated.

**Pull request merged**